services/shared/src/main/java/com/wedriveu/services/shared/rabbitmq/VerticlePublisher.java

```diff
@@ -0,0 +1,57 @@
+package com.wedriveu.services.shared.rabbitmq;
+
+import com.wedriveu.services.shared.utilities.Log;
+import io.vertx.core.AbstractVerticle;
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.core.json.JsonObject;
+import io.vertx.rabbitmq.RabbitMQClient;
+
+import static com.wedriveu.services.shared.utilities.Constants.EXCHANGE_TYPE;
+import static com.wedriveu.services.shared.utilities.Constants.VEHICLE_SERVICE_EXCHANGE;
+
+/**
+ * Created by Marco on 07/08/2017.
+ */
+public class VerticlePublisher extends AbstractVerticle {
+
+    private static RabbitMQClient client;
+    private static String TAG = VerticlePublisher.class.getSimpleName();
+
+    @Override
+    public void start(Future<Void> startFuture) throws Exception {
+        super.start(startFuture);
+        startFuture.complete();
+    }
+
+    protected void publish(String exchangeName, String routingKey, JsonObject data) {
+        client = RabbitMQConfig.getInstance(vertx).getRabbitMQClient();
+        client.start(onStartCompleted -> {
```

Comment:
So, a publish is not only a publish to rabbitmq, but also a starte and a declare exchange? Are not too much responsibilities for one method?